summingbird-storm/src/main/scala/com/twitter/summingbird/storm/SpoutProvider.scala

```diff
@@ -0,0 +1,152 @@
+package com.twitter.summingbird.storm
+
+import backtype.storm.metric.api.IMetric
+import backtype.storm.task.TopologyContext
+import backtype.storm.topology.{ IRichSpout, TopologyBuilder }
+import com.twitter.algebird.Semigroup
+import com.twitter.algebird.util.summer.Incrementor
+import com.twitter.summingbird._
+import com.twitter.summingbird.batch.{ BatchID, Timestamp }
+import com.twitter.summingbird.online.{ Externalizer, executor }
+import com.twitter.summingbird.online.option.SummerBuilder
+import com.twitter.summingbird.option.JobId
+import com.twitter.summingbird.planner.{ Dag, SummerNode }
+import com.twitter.summingbird.storm.planner.StormNode
+import com.twitter.tormenta.spout.Spout
+import com.twitter.summingbird.storm.Constants
+import com.twitter.summingbird.storm.option.SpoutStormMetrics
+import com.twitter.summingbird.storm.spout.KeyValueSpout
+
+case class SpoutProvider(storm: Storm, stormDag: Dag[Storm], node: StormNode, jobID: JobId)(implicit topologyBuilder: TopologyBuilder) {
+
+  private def getOrElse[T <: AnyRef: Manifest](stormDag: Dag[Storm], queryNode: StormNode = node, default: T) = storm.getOrElse(stormDag, queryNode, default)
+
+  private val metrics = getOrElse(stormDag, node, Constants.DEFAULT_SPOUT_STORM_METRICS)
+
+  private def computerSpout(): (Int, Spout[(Timestamp, Any)]) = {
```

Comment:
The parallelism (`Int`) and `Spout` aren't really entangled and shouldn't be constructed by a single method.  We can have:
```scala
def spoutParallelism: Int =...
def tormentaSpout: Spout[(Timestamp, Any)] = ...
```
(I'm not sure `computerSpout` is an intuitive name)